### PR TITLE
New spells

### PR DESCRIPTION
**Pull request by daniela-venuta**

### CODE REVIEW

**Review by redknight990 — APPROVED** [review]
Looks Good! Nice rework to accommodate multiple spell types!

**Review by arejayelle — COMMENTED** [review]
Fireball light seems to go at a 90ish degree angle from where the fireball actually is + there are bugs w/ the invisiblity spell (as discussed)

**Review by arejayelle — APPROVED** [review]
Would prefer to have OnStart instead of onAwake for the spell initialization but since this is gonna be replaced SOON ish, it's fine. 

Don't forget to make an issue for the fireball bug

**Comment by daniela-venuta** [comment]
> Fireball light seems to go at a 90ish degree angle from where the fireball actually is + there are bugs w/ the invisiblity spell (as discussed)

Created a bug for fireball and invisibility has been fixed

File: Hush/Assets/Scripts/Weapon/StunSpell.cs [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        private static InputManager Input => InputManager.Instance;
+        private static SpellManager SpellManager => SpellManager.Instance;
+
+        private PlayerMovement _player;
+        
+        private void Awake()
+        {
+             //GameManager.Instance.SetActiveHeavySpell(SpellType.StunSpell);
+             //GameManager.Instance.SetActiveLightSpell(SpellType.StunSpell);
+        }
```
Comment (po-trottier) [inline Hush/Assets/Scripts/Weapon/StunSpell.cs:36]:
```suggestion
```

File: Hush/Assets/Scripts/Weapon/Enums/StunCollision.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+using System.Collections.Generic;
+using Common.Enums;
+using Enemies;
+
+namespace Weapon.Enums
+{
+    public class StunCollision : MonoBehaviour
+    {
+        public float duration = 1f;
+        void OnParticleCollision(GameObject other)
```
Comment (po-trottier) [inline Hush/Assets/Scripts/Weapon/Enums/StunCollision.cs:12]: Might wanna use a SphereCast in the Spell Prefab to do that

**Review by po-trottier — CHANGES_REQUESTED** [review]
Good job! Few small changes

**Comment by po-trottier** [comment]
> > Fireball light seems to go at a 90ish degree angle from where the fireball actually is + there are bugs w/ the invisiblity spell (as discussed)
> 
> Created a bug for fireball and invisibility has been fixed

Fireball bug was already fixed by @redknight990 in his PR

**Review by po-trottier — APPROVED** [review]
Gucci

**Review by po-trottier — APPROVED** [review]
Good again

**Pull request merged**